src/analyze-action.ts

```diff
@@ -74,22 +74,24 @@ async function sendStatusReport(
     error?.message,
     error?.stack,
   );
-  const report: FinishStatusReport = {
-    ...statusReportBase,
-    ...(stats || {}),
-    ...(dbCreationTimings || {}),
-  };
-  if (config && didUploadTrapCaches) {
-    const trapCacheUploadStatusReport: FinishWithTrapUploadStatusReport = {
-      ...report,
-      trap_cache_upload_duration_ms: Math.round(trapCacheUploadTime || 0),
-      trap_cache_upload_size_bytes: Math.round(
-        await getTotalCacheSize(config.trapCaches, logger),
-      ),
+  if (statusReportBase !== undefined) {
+    const report: FinishStatusReport = {
+      ...statusReportBase,
+      ...(stats || {}),
+      ...(dbCreationTimings || {}),
     };
-    await statusReport.sendStatusReport(trapCacheUploadStatusReport);
-  } else {
-    await statusReport.sendStatusReport(report);
+    if (config && didUploadTrapCaches) {
```

Comment:
Do we ever reasonably expect config to be falsy, but didUploadTrapCaches to be `true`? If so, we might want to still upload in that case:

```suggestion
    if (didUploadTrapCaches) {
```

And the change below. But only if this is not already an error state.

Reply:
I don't think so — if `config` is `undefined`, we should be erroring out much earlier: https://github.com/github/codeql-action/blob/f421cda8e715cc7652d8b748f49574cf286aec12/src/analyze-action.ts#L205.